### PR TITLE
Implemented Spark item to item recommenders

### PR DESCRIPTION
**Pull request by ChuyangKe**
### Description

Implemented get_topk_most_similar_users() and get_popularity_based_topk() for SAR+.

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: contrib/sarplus/python/pysarplus/SARPlus.py [hunk 1]
```diff
@@ -488,3 +524,72 @@ def recommend_k_items(
             )
         else:
             raise ValueError("No cache_path specified")
+
+    def get_topk_most_similar_users(self, test, user, top_k=10):
```
Comment (miguelgfierro) [inline contrib/sarplus/python/pysarplus/SARPlus.py:528]: One question, have you checked that the results between the CPU version and this one are the same?

File: contrib/sarplus/python/pysarplus/SARPlus.py [hunk 2]
```diff
@@ -171,42 +180,54 @@ def fit(self, df):
...
         item_cooccurrence = self.spark.sql(query)
         item_cooccurrence.write.mode("overwrite").saveAsTable(
             self._format("{prefix}item_cooccurrence")
         )
 
+        # compute item frequencies
+        self.item_frequencies = item_cooccurrence.filter(
+            F.col("i1") == F.col("i2")
+        ).select(F.col("i1").alias("item_id"), F.col("value").alias("frequency"))
+
```
Comment (miguelgfierro) [inline contrib/sarplus/python/pysarplus/SARPlus.py:195]: question to @simonzhaoms will the item-item function have any relationship with the work you are doing about the new similarities?

File: contrib/sarplus/python/pysarplus/SARPlus.py [hunk 3]
```diff
@@ -488,3 +524,72 @@ def recommend_k_items(
...
+            )
+            .groupBy("matrix.user_id")
+            .agg(F.sum("prod").alias("similarity"))
+            .orderBy("similarity", ascending=False)
+            .limit(k)
+        )
+
+        return df_similar_users
+
+    def get_popularity_based_topk(self, top_k=10, items=True):
```
Comment (miguelgfierro) [inline contrib/sarplus/python/pysarplus/SARPlus.py:579]: are you planning to do the test on this PR or in a new one?

File: contrib/sarplus/python/pysarplus/SARPlus.py [hunk 4]
```diff
@@ -488,3 +524,72 @@ def recommend_k_items(
...
+        Returns:
+            pyspark.sql.DataFrame: Spark dataframe with top k most popular items
+            and their frequencies in descending order.
+        """
+
+        # TODO: get most frequent users
+        if not items:
+            raise ValueError("Not implemented")
+
+        return self.item_frequencies.orderBy("frequency", ascending=False).limit(top_k)
```
Comment (miguelgfierro) [inline contrib/sarplus/python/pysarplus/SARPlus.py:595]: I noticed that the sarplus tests are triggered, but there is a badge in the main page that is failing: https://github.com/microsoft/recommenders/tree/staging/contrib/sarplus @simonzhaoms do you know what is the problem?

File: contrib/sarplus/python/pysarplus/SARPlus.py (hunk 1 above)
Comment (ChuyangKe) [reply contrib/sarplus/python/pysarplus/SARPlus.py:528]: Yes, we are using the same test cases as for the CPU version.

**Pull request merged**